### ISSUE TITLE
Hubs loads into a black screen

### ISSUE THREAD

**Issue by gfodor** [MozillaReality/FirefoxReality#545]
## Hardware
Oculus Go

## Steps to Reproduce
1. go to hubs.mozilla.com
2. create a room with the "atrium" space
3. hit "enter in VR" on the landing page

## Current Behavior
black screen (may not be 100% of the time)

## Expected Behavior
you should enter the scene in 3d

## Possible Solution
this may be related to shaders we are using in the scene. the "wide open space" scene may work for comparison

```
```

</details>

**Comment by bluemarvin** [comment]
If you hit the back button, you will see the room in the browser window now. If you try and enter VR from the browser window you will then be able to enter immersive mode. At least that worked for me. Not sure why entering the first time fails to draw.

**Comment by cvan** [comment]
I have seen this quite frequently lately across many WebVR sites. I have not gone deep with debugging yet. if I find anything, I shall post my findings here.

**Comment by avrignaud** [comment]
How can we dive a bit deeper and explore? Really should have Hubs just work in FxR... :)

**Comment by bluemarvin** [comment]
@MortimerGoro maybe if you have some time you can investigate the black screen?

**Comment by MortimerGoro** [comment]
I found that the black screen happens when the canvas is not composited before entering VR. I also was able to reproduce the problem with the "wide open space" that @gfodor thought that worked ok. 

The canvas is not composited in this 2 situations:

- I created a local test project. If we don't add a canvas to the DOM and use it in VR, FxR always renders a black screen. I tried the same on Oculus Go and it works ok. 
- In Hubs (or WebVR link traversal) there is a race condition in which the canvas is not composited. When we enter VR we pause the FxR compositor. As Hubs (or link traversal) doesn't show the final canvas in a WebGL 2D scene, the change of this issue happening is higher

I'm going to try to find a solution in Gecko for the problem. In the worst case we can delay the compositor pause a bit to fix the problem in Hubs or Link traversal. But the good solution is that a canvas that is never added to the DOM should also work.
